tools/flang1/flang1exe/semant.c

```diff
@@ -6183,7 +6183,7 @@ semant1(int rednum, SST *top)
         sem.bounds[sem.arrdim.ndim].lwast = ast;
       }
     }
-    if (constarraysize && arraysize <= 0) {
+    if (constarraysize && arraysize < 0) {
```

Comment:
@bryanpkc can you explain the rationale behind that change? Looks like now flang has stopped to warn about zero-sized arrays. There is a nag test (z2a.f90) which expect a compiler to warn about zero sized array and now this test fails as flang accepts such array silently.

Reply:
@pawosm-arm I believe this was done in our downstream compiler to make its behavior consistent with both gfortran and ifort, which do not produce any error when an C-interoperable array has a size of zero (https://godbolt.org/z/Pe1bMxdzz), despite the wording in the Fortran 2008 specification (section 15.3.6, *Interoperability of array variables*). This change should not have been included in this PR, and I apologize for the mistake. I'll create a PR to revert this change, or perhaps more appropriately, limit the check to only interoperable arrays.